x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -526,7 +526,7 @@ public void onStackTraceResponse(MultiGetResponse multiGetItemResponses) {
                         if (stackTracePerId.putIfAbsent(id, stacktrace) == null) {
                             totalFrames.addAndGet(stacktrace.frameIds.size());
                             stackFrameIds.addAll(stacktrace.frameIds);
-                            executableIds.addAll(stacktrace.fileIds);
+                            stacktrace.forNativeAndKernelFrames(executableId -> executableIds.add(executableId));
```

Comment:
Nit: I'd probably just use `e` instead of `executableId`. Usually the lambda parameter is clear from the context:

 ```suggestion
                            stacktrace.forNativeAndKernelFrames(e -> executableIds.add(e));
```

I leave it up to you what you prefer.